doc/vrrp/VRRP_Adaptation_HLD.md

```diff
@@ -0,0 +1,712 @@
+state           = vrrp_state                    ; String denoting the state of VRRP instance
+version         = vrrp_version                  ; VRRP version. Value is 2 or 3. Default is 3
+preempt         = "enabled"/"disabled"          ; VRRP pre-emption is enabled? Default is enabled
+track_interface = interface_Name|weight_value   ;This is repeated for the configured tracking interfaces
+```
+Example:-
+```
+admin@sonic:~$ redis-cli -n 4 keys VRRP*
+
+"VRRP|Ethernet8|8"
+"VRRP|Ethernet1|1"
+"VRRP|Vlan10|10"
+"VRRP|Ethernet2|2"
+"VRRP|Vlan4|4"
+
+admin@sonic:~$ redis-cli -n 4 HGETALL " VRRP|Vlan10|10"
+
+"vrid"
+"10"
+"vip"
+"4.1.1.100/24,4.1.1.200/24,2000::50/64"
+"priority"
+"100"
+"adv_interval"
+"1000"
+"state"
+""
+"version"
+"3"
+"preempt"
+"enabled"
+"track_interface"
+"Ethernet5|10, Ethernet9|20"
+```
+
+#### APPL_DB Changes
+
+VRRP_TABLE
+
+Producer: vrrpmgrd and vrrpsyncd
+
+Consumer: vrrporch
+
+Description: This is a new table that contains VRRP state information. This entry will be added to APP_DB for each VRRP instance.
+```
+; New table
+; holds the VRRP state and VMAC information
+
+key         = VRRP_TABLE:interface_name:vrid:type
+              interface_name        ; interface name as a string. Vlan, Ethernet or PortChannel
+              vrid                  ; vrid is an integer
+              type                  ; IP address type string. IPv4 or IPv6
+    
+; field = value
+vmac        = virtual_mac_address   ; Virtual MAC address associated with VRRP instance
+state       = "up" | "down"         ; Linux macvlan device state
+priority    = vrrp_priority         ; Priority of VRRP instance
+```
+Example:-
+```
+admin@sonic:~$ redis-cli -n 0 HGETALL "VRRP_TABLE:Ethernet36:6:IPv4"
+1) "vmac"
+2) "00:00:5e:00:01:06"
+3) "state"
+4) "up"
+5) "priority "
+6) "100"
+```
+
+#### ASIC_DB Changes
+
+Example:-
+```
+admin@sonic:~$ redis-cli -n 1 hgetall "ASIC_STATE:SAI_OBJECT_TYPE_ROUTER_INTERFACE:oid:0x6000000000a80"
+ 1) "SAI_ROUTER_INTERFACE_ATTR_VIRTUAL_ROUTER_ID"
+ 2) "oid:0x300000000003a"
+ 3) "SAI_ROUTER_INTERFACE_ATTR_SRC_MAC_ADDRESS"
+ 4) "00:00:5E:00:01:06"
+ 5) "SAI_ROUTER_INTERFACE_ATTR_TYPE"
+ 6) "SAI_ROUTER_INTERFACE_TYPE_PORT"
+ 7) "SAI_ROUTER_INTERFACE_ATTR_PORT_ID"
+ 8) "oid:0x1000000000025"
+ 9) "SAI_ROUTER_INTERFACE_ATTR_IS_VIRTUAL"
+10) "true"
+```
+
+#### Modules Design and Flows
+
+##### ADD/DEL VRRP instance
+
+![VRRP ADD/DEL VRRP instance Flows](images/VRRP_Config_Instance_Flow.png)
+
+vrrpcfgd:
+  - Listens to VRRP create, delete and parameter change in CONFIG DB
+  - Update changes by using FRR CLI commands
+
+vrrpd:
+  - Listens to VRRP create, delete and parameter change in vrrpcfgd
+  - Run VRRP state machine
+    - For VRRP instances state transition to Master, set the Macvlan device into protodown off
+    - For VRRP instances state transition to non-Master, set the Macvlan interface into protodown on
+
+vrrpmgrd:
```

Comment:
vrrpmgrd, vrrpsyncd, vrrporch should be moved into swss container.

Reply:
OK.